### PR TITLE
Allow selecting the zone used by the python wrapper (for multizone drivers)

### PR DESCRIPTION
**Pull request by afshawnlotfi**
## Proposed Changes
Added `SelectZone` function to `CDriverBase` that dynamically sets the zone index
 

## Example Usage
```python
import pysu2
from mpi4py import MPI

def execute_su2():
    # Import mpi4py for parallel run
    comm = MPI.COMM_WORLD
    rank = comm.Get_rank()
    num_zones = 2
    # Initialize the corresponding driver of SU2, this includes solver preprocessing
    SU2Driver = pysu2.CMultizoneDriver(config_path, num_zones, comm) # type: ignore

    # Get all the markers defined on this rank and their associated indices.
    allMarkerIDs = SU2Driver.GetMarkerIndices()

    # Time loop is defined in Python so that we have acces to SU2 functionalities at each time step
    comm.Barrier()

    # Time iteration preprocessing
    SU2Driver.Preprocess(0)

    # Run one time-step (static: one simulation)
    SU2Driver.Run()

    # Update the solver for the next time iteration
    SU2Driver.Update()

    # Monitor the solver and output solution to file if required
    SU2Driver.Monitor(0)

    for izone in range(num_zones):
        SU2Driver.SelectZone(izone)
        # Get all the markers defined on this rank and their associated indices.
        allMarkerIDs = SU2Driver.GetMarkerIndices()
        print(allMarkerIDs)

    # Output the solution to file
    SU2Driver.Output(0)

    # Finalize the solver and exit cleanly
    SU2Driver.Finalize()
```

## PR Checklist
*Put an X by all that apply. You can fill this out after submitting the PR. If you have any questions, don't hesitate to ask! We want to help. These are a guide for you to know what the reviewers will be looking for in your contribution.*

- [X] I am submitting my contribution to the develop branch.
- [X] My contribution generates no new compiler warnings (try with --warnlevel=3 when using meson).
- [X] My contribution is commented and consistent with SU2 style (https://su2code.github.io/docs_v7/Style-Guide/).
- [X] I have added a test case that demonstrates my contribution, if necessary.
- [X] I have updated appropriate documentation (Tutorials, Docs Page, config_template.cpp), if necessary.

### CODE REVIEW

File: SU2_CFD/src/drivers/CMultizoneDriver.cpp [hunk 1]
```diff
@@ -248,7 +248,7 @@ void CMultizoneDriver::Preprocess(unsigned long TimeIter) {
   }
 
   /*--- Ramp turbo values for unsteady problems here, otherwise do it over outer iterations. ---*/
-  if (config_container[ZONE_0]->GetTime_Domain()) {
```
Comment (pcarruscag) [inline SU2_CFD/src/drivers/CMultizoneDriver.cpp:251]: You need to revert all the changes in this file

File: SU2_CFD/src/drivers/CDriver.cpp [hunk 2]
```diff
@@ -120,8 +120,8 @@ CDriverBase(confFile, val_nZone, MPICommunicator), StopCalc(false), fsi(false),
 
   /*--- Retrieve dimension from mesh file ---*/
 
-  nDim = CConfig::GetnDim(config_container[ZONE_0]->GetMesh_FileName(),
-                          config_container[ZONE_0]->GetMesh_FileFormat());
+  nDim = CConfig::GetnDim(config_container[selected_iZone]->GetMesh_FileName(),
+                          config_container[selected_iZone]->GetMesh_FileFormat());
```
Comment (pcarruscag) [inline SU2_CFD/src/drivers/CDriver.cpp:124]: And also all the changes in this file

**Review by pcarruscag — COMMENTED** [review]
Thanks for taking the lead on this, you changed too many places, only the functions of the python wrapper (the ones in the pySU2 group) need to be changed.
The others are internal functions of the drivers that need to work on the zone the driver decides.

**Comment by pcarruscag** [comment]
@afshawnlotfi is this working as expected?

File: SU2_CFD/include/drivers/CDriverBase.hpp [hunk 3]
```diff
@@ -54,7 +54,7 @@ class CDriverBase {
       UsedTime;        /*!< \brief Elapsed time between Start and Stop point of the timer. */
 
   unsigned long TimeIter;
-
+  unsigned long selected_iZone = ZONE_0; /*!< \brief Selected zone for the driver. Defaults to ZONE_0 */
```
Comment (pcarruscag) [inline SU2_CFD/include/drivers/CDriverBase.hpp:57]:
```suggestion
  unsigned short selected_iZone = ZONE_0; /*!< \brief Selected zone for the driver. Defaults to ZONE_0 */
```

File: SU2_CFD/include/drivers/CDriverBase.hpp [hunk 4]
```diff
@@ -530,6 +530,17 @@ class CDriverBase {
     main_geometry->SetCustomBoundaryHeatFlux(iMarker, iVertex, WallHeatFlux);
   }
 
+
+  /*!
+   * \brief Selects zone to be used for Driver operation
+   * \param[in] iZone - Zone identifier.
+   */
+  inline void SelectZone(unsigned long iZone) {
```
Comment (pcarruscag) [inline SU2_CFD/include/drivers/CDriverBase.hpp:538]:
```suggestion
  inline void SelectZone(unsigned short iZone) {
```

File: SU2_CFD/include/drivers/CDriverBase.hpp [hunk 5]
```diff
@@ -530,6 +530,17 @@ class CDriverBase {
     main_geometry->SetCustomBoundaryHeatFlux(iMarker, iVertex, WallHeatFlux);
   }
 
+
```
Comment (pcarruscag) [inline SU2_CFD/include/drivers/CDriverBase.hpp:533]:
```suggestion
```

File: SU2_CFD/include/drivers/CDriverBase.hpp [hunk 6]
```diff
@@ -530,6 +530,17 @@ class CDriverBase {
...
+  /*!
+   * \brief Selects zone to be used for Driver operation
+   * \param[in] iZone - Zone identifier.
+   */
+  inline void SelectZone(unsigned long iZone) {
+    selected_iZone = iZone;
```
Comment (pcarruscag) [inline SU2_CFD/include/drivers/CDriverBase.hpp:539]:
```suggestion
    if (iZone >= nZone) SU2_MPI::Error("Zone index out of range", CURRENT_FUNCTION);
    selected_iZone = iZone;
```

**Review by pcarruscag — APPROVED** [review]

**Comment by afshawnlotfi** [comment]
> @afshawnlotfi is this working as expected?

Sorry this last week has been busy, will verify by the end of this week.

**Pull request merged**